packages/replayio/src/utils/recordings/createdDeferredAction.ts

```diff
@@ -0,0 +1,19 @@
+import { createDeferred } from "../createDeferred";
+import { debug } from "./debug";
+
+export function createdDeferredAction<Data>(data: Data, promise: Promise<void>) {
```

Comment:
this is somewhat similar to `Promise.allSettled`. WDYT about `createSettledDeferred`?

Reply:
I don't think this is actually very similar to `allSettled` so I think the rename probably wouldn't make sense. (We can keep chatting about this though if you think I'm wrong!)

Reply:
I see the "settled" part to be quite similar here. `createDeferred` gets some mutable data, creates the promise and gives the user control over its resolution etc. This is `Promise.withResolvers` with some extra things on top of it.

The `createdDeferredAction` wraps the above with one extra thing - it eliminates the rejected state altogether from your async action. We want to be able to track failure and all but we don't want to fail hard on failures. This is quite similar to the "settled" part - it just uses a boolean to denote its result rather than `{ status: 'fulfilled', value: T } | { status: 'rejected', reason: unknown }`. We could easily use the "standardized" shape instead of a boolean here - but both are fine.

One extra thing is that this still gives the caller control over the deferred itself. This calls `.resolve`/`.reject` but both are still returned to the caller and they could, accidentally, call them and settle at some other time.

Anyway, I'm just saying where I've seen the similarity here.

Reply:
I think I understand the similarity you're describing– specifically with the resolve/reject status. That's not a core aspect of `createDeferred` though. It's just how I used it in this _one specific case_. In most cases (there are a lot within the DevTools codebase) we resolve/reject just like you would with a Promise. The key difference being that deferred also provides a way to synchronously read a rejected/resolved value (without the microtask loop).

Reply:
Yeah, so I didn't mean to change anything about `createDeferred` - just this one wrapper here (`createdDeferredAction`) that has a more specific use case

Reply:
Oh! I totally misunderstood the scope of what you were suggesting. Yeah okay let me re-think about that.